File: MachineProblem2.m
p1=input('Input point 1 in [x y] form:');
p2=input('Input point 2 in [x y] form:');
p3=input('Input point 3 in [x y] form:');

A=[p1(1) p1(2) 1; p2(1) p2(2) 1; p3(1) p3(2) 1]; %coefficients of x and y
B=[-(p1(1)^2+p1(2)^2); -(p2(1)^2+p2(2)^2); -(p3(1)^2+p3(2)^2)]; %constants at right hand side
C=inv(A)*B; %solution to equations A*[x y]=b; [x y]=b*A^-1;

D=C(1);
E=C(2);
F=C(3);
h=-D/2;k=-E/2; %formula to get h, k values
radius=sqrt((D^2+E^2-4*F)/(4)); % formula to get radius

coefficientsDEF=[D E F]
centerhk=[h k]
radius

%fprintf('D=%f\t',D);
%fprintf('E=%f\t\t',E);
%fprintf('F=%f\n',F);
%fprintf('h=%f\t\t',h);
%fprintf('k=%f\n',k);
%fprintf('r=%f\n',r);

File: MachineProblem2.py
import numpy as np

x1, y1 = map(int, input("Input the first point: ").split())
x2, y2 = map(int, input("Input the second point: ").split())
x3, y3 = map(int, input("Input the third point: ").split())

# Coefficients of variable X for the systems of linear equations
A = np.array([[x1, y1, 1], [x2, y2, 1], [x3, y3, 1]])
# Constants at the right-hand side.
B = np.array([[-(x1**2 + y1**2), -(x2**2 + y2**2), -(x3**2 + y3**2)]])
B = np.transpose(B)
# Solutions to the systems of equations
C = np.linalg.inv(A).dot(B)

# Computation for center (h,k) and radius 
h = -C[0]/2
k = -C[1]/2
r = np.sqrt((C[0]**2 + C[1]**2 - 4*C[2])/4)

print('The center is at ', h, k)
print('The radius of the circle is ', r)
print('The coefficients D, E, F of the general equation are ', C) 